### PR TITLE
AI proxy (+ Advanced)

### PR DESCRIPTION
**Pull request by lmilan**
### Preview Links
* https://deploy-preview-175--kongdeveloper.netlify.app/plugins/ai-proxy-advanced/

### Checklist 

- [x] Every page is page one
- [x] Tested how-to docs. If not, note why here. 
- [x] All pages contain metadata
- [ ] Updated [sources.yaml](https://github.com/Kong/developer.konghq.com/blob/main/tools/track-docs-changes/config/sources.yml). For more info review [track docs changes](https://github.com/Kong/developer.konghq.com/tree/main/tools/track-docs-changes)
- [ ] Any new docs link to existing docs.

### CODE REVIEW

File: app/_data/plugins/ai-proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    completions:
+      supported: ✅
+      upstream_path: '/v1/complete'
+      route_type: 'llm/v1/completions'
+      model_example: 'claude-2.1'
+    streaming:
+      supported: Chat type only
+  - name: 'Mistral'
+    formats: 'mistral.ai, OpenAI, raw, and OLLAMA formats'
+    url_pattern: 'As defined in `{{ upstream_url }}'
```
Comment (Guaris) [inline app/_data/plugins/ai-proxy.yaml:65]: Should this look like this 
image

File: app/_includes/plugins/ai-proxy/formats.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+{% include plugins/ai-proxy/tables/upstream-urls.html providers=providers %}
+
+{:.important}
+> While only the **Llama2** and **Mistral** models are classed as self-hosted, the target URL can be overridden for any of the supported providers.
+> For example, a self-hosted or otherwise OpenAI-compatible endpoint can be called by setting the same [`{{ upstream_url }}`](/plugins/{{ id }}/reference/#{{ upstream_url_slug }}) plugin option.
+
+### Input formats
+Kong will mediate the request and response format based on the selected [`{{ provider }}`](/plugins/{{ id }}/reference/#{{ provider_slug }}) and [`{{ route_type }}`](/plugins/{{ id }}/reference/#{{ route_type_slug }}), as outlined in the table above.
+
+The Kong AI Proxy accepts the following inputs formats, standardized across all providers; the `{{ route_type }}` must be configured respective to the required request and response format examples:
```
Comment (Guaris) [inline app/_includes/plugins/ai-proxy/formats.md:33]:
```suggestion
The Kong AI Proxy accepts the following inputs formats, standardized across all providers. The `{{ route_type }}` must be configured respective to the required request and response format examples:
```

File: app/_includes/plugins/ai-proxy/links.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+* Learn how to use the plugin with different providers:
+  * [OpenAI](/plugins/{{ id }}/examples/openai-chat-route/)
+  * [Cohere](/plugins/{{ id }}/examples/cohere-chat-route/)
+  * [Azure](/plugins/{{ id }}/examples/azure-chat-route/)
+  * [Anthropic](/plugins/{{ id }}/examples/anthropic-chat-route/)
+  * [Mistral](/plugins/{{ id }}/examples/mistral-chat-route/)
+  * [Llama2](/plugins/{{ id }}/examples/llama2-chat-route/)
+  * [Gemini/Vertex](/plugins/{{ id }}/examples/gemini-chat-route/)
+  * [Amazon Bedrock](/plugins/{{ id }}/examples/bedrock-chat-route/)
+  * [Hugging Face](/plugins/{{ id }}/examples/huggingface-chat-route/)
```
Comment (Guaris) [inline app/_includes/plugins/ai-proxy/links.md:16]:
```suggestion
## Get started
* [Configuration reference](/plugins/{{ id }}/reference/)
* [Configuration examples](/plugins/{{ id }}/examples/)

File: app/_includes/plugins/ai-proxy/tables/supported-providers.html
Comment (Guaris) [inline app/_includes/plugins/ai-proxy/tables/supported-providers.html:1]: What if this lived on a separate provider page. Then we can reference it within the specific plugin page and link out to the provider page?

Comment (Guaris) [reply app/_includes/plugins/ai-proxy/tables/supported-providers.html:1]: I would do this next year not now.

**Comment by Guaris** [comment]
Add docs.konghq.com links to the broken links and lets just ship this and come back.

**Review by Guaris — APPROVED** [review]

File: app/_data/plugins/ai-proxy.yaml (hunk 1 above)
Comment (lmilan) [reply app/_data/plugins/ai-proxy.yaml]: I fixed it, the way I did it is not ideal but we're going to revisit the whole thing later anyway, so I think it's good enough for now

**Pull request merged**